dev/NavigationView/NavigationView.cpp

```diff
@@ -2944,9 +2944,10 @@ void NavigationView::UpdatePaneDisplayMode(winrt::NavigationViewPaneDisplayMode
 
     UpdatePaneDisplayMode();
 
-    // For better user experience, We help customer to Open/Close Pane automatically when we switch between LeftMinimal <-> other PaneDisplayMode.
+    // For better user experience, We help customer to Open/Close Pane automatically when we switch between LeftMinimal <-> Left.
     // From other navigation PaneDisplayMode to LeftMinimal, we expect pane is closed.
-    // From LeftMinimal to other left PaneDisplayMode other than Auto, we expect Pane is opened.
+    // From LeftMinimal to Left, it is expected the pane is open. For other configurations, this seems counterintuitive.
+    // See #1702 and #1787
```

Comment:
nit: Just a thought if the code below would make it easier to read the code.

if(IsPaneOpen())
{
    if(newDisplayMode == LeftMinimal)
    {
       ClosePane();
    }
}
else
{
    if(oldDisplayMode == LeftMinimal && newDisplayMode = Left)
    {
      OpenPane();
    }
}

Reply:
Yes that should work. I will update my code as soon as I have time.